provider/mesh/peer.go

```diff
@@ -0,0 +1,273 @@
+package mesh
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+	"github.com/satori/go.uuid"
+	"github.com/weaveworks/mesh"
+)
+
+// NotificationInfos provides gossiped information about which
+// receivers have been notified about which alerts.
+type NotificationInfos struct {
+	st     *notificationState
+	send   mesh.Gossip
+	logger log.Logger
+}
+
+func NewNotificationInfos(logger log.Logger) *NotificationInfos {
```

Comment:
On 22 June 2016 at 09:39, Fabian Reinartz <notifications@github.com> wrote:

> Yea, I'll add respective comments. I still find it utterly useless and a
> tool that forces me to do useless things doesn't seem like a particularly
> good tool. After all the tool should help me, not the other way around.


Thanks for adding the doc comments.

The tool correctly implements the style guide of the Go community, which we
adopted for Prometheus. The rule is that every exported identifier requires
a doc comment. Obviously, we could decide to change the style guide, but
based on my personal experience, I predict the debate will be moot.  (I had
that debate quite often for different languages, taking either side at
times, but the outcome never changed.)

-- 
Björn Rabenstein, Engineer
http://soundcloud.com/brabenstein

SoundCloud Ltd. | Rheinsberger Str. 76/77, 10115 Berlin, Germany
Managing Director: Alexander Ljung | Incorporated in England & Wales with
Company No. 6343600 | Local Branch Office | AG Charlottenburg  | HRB 110657B